src/lib/components/Profile.svelte

```diff
@@ -1,117 +1,44 @@
 <script lang="ts">
 	import type { Profile } from '$lib/types/profile';
-	import { enhance } from '$app/forms';
-	import { v4 as uuidv4 } from 'uuid';
-
-	import { supabase } from '$lib/auth';
-	import { invalidate } from '$app/navigation';
 
 	import Icon from '$lib/components/Icon.svelte';
-	import Modal from '$lib/components/Modal.svelte';
-	import Button from '$lib/components/Button.svelte';
-	import Input from '$lib/components/forms/Input.svelte';
-	import Checkbox from '$lib/components/forms/Checkbox.svelte';
 	import Curriculum from './Curriculum.svelte';
 	import Portfolio from './Portfolio.svelte';
 
-	let files: FileList;
-	import { uploadImg, uploading } from '$lib/utils/upload';
-
-	export let editable = false;
-	export let userId: string;
 	export let profile: Profile;
-	export let form: { first_name_error: string; last_name_error: string } | null = null;
+	export let openEditModal: null | (() => void) = null;
```

Comment:
It might be simpler to have these functions defined here directly rather than passed as props.